tests/Gaufrette/Functional/Adapter/AwsS3Test.php

```diff
@@ -2,116 +2,144 @@
 
 namespace Gaufrette\Functional\Adapter;
 
-use Gaufrette\Adapter\AwsS3;
 use Aws\S3\S3Client;
-use Guzzle\Plugin\Mock\MockPlugin;
-use Guzzle\Http\Message\Response;
+use Gaufrette\Adapter\AwsS3;
+use Gaufrette\Filesystem;
 
 /**
  * @todo move to phpspec
  */
 class AwsS3Test extends \PHPUnit_Framework_TestCase
 {
-    protected function getClient()
+    static private $SDK_VERSION = 3;
+
+    /** @var string */
+    private $bucket;
+
+    /** @var S3Client */
+    private $client;
+
+    public static function setUpBeforeClass()
     {
-        return S3Client::factory(array(
-            'key'    => 'foo',
-            'secret' => 'bar'
-        ));
+        $installed = json_decode(file_get_contents(__DIR__.'/../../../../vendor/composer/installed.json'), true);
+        $sdk = current(array_filter($installed, function ($dependency) {
+            return $dependency['name'] === 'aws/aws-sdk-php';
+        }));
+
+        if (version_compare($sdk['version'], '3.0.0') === -1) {
+            self::$SDK_VERSION = 2;
+        }
     }
 
-    public function testCreatesBucketIfMissing()
+    public function setUp()
     {
-        $mock = new MockPlugin(array(
-            new Response(404),                // Head bucket response
-            new Response(200),                // Create bucket response
-            new Response(200, array(), 'foo') // Get object response
-        ));
-        $client = $this->getClient();
-        $client->addSubscriber($mock);
-        $adapter = new AwsS3($client, 'bucket', array('create' => true));
-        $this->assertEquals('foo', $adapter->read('foo'));
-
-        $requests = $mock->getReceivedRequests();
-        $this->assertEquals('HEAD', $requests[0]->getMethod());
-        $this->assertEquals('PUT', $requests[1]->getMethod());
-        $this->assertEquals('GET', $requests[2]->getMethod());
-        $this->assertEquals('bucket.s3.amazonaws.com', $requests[0]->getHost());
+        $key = getenv('AWS_KEY');
+        $secret = getenv('AWS_SECRET');
+
+        if (empty($key) || empty($secret)) {
+            $this->markTestSkipped();
+        }
+
+        $this->bucket = uniqid(getenv('AWS_BUCKET'));
+
+        if (self::$SDK_VERSION === 3) {
```

Comment:
isn't there a way to feature-detect it instead of using the composer metadata ? Maybe with ``method_exists(S3Client::class, 'factory')`` to detect version 2 ?